java/ql/src/semmle/code/java/dataflow/ExternalFlow.qll

```diff
@@ -679,13 +679,27 @@ private predicate summaryElementRef(Top ref, string input, string output, string
   )
 }
 
+bindingset[name]
+private Field interpretField(string name) {
+  exists(string splitRegex, string package, string className, string fieldName |
+    splitRegex = "^(.*)\\.([^.]+)\\.([^.]+)$" and
+    package = name.regexpCapture(splitRegex, 1) and
+    className = name.regexpCapture(splitRegex, 2) and
+    fieldName = name.regexpCapture(splitRegex, 3)
+  |
+    result = any(Field f | f.hasQualifiedName(package, className, fieldName))
+  )
+}
```

Comment:
Instead of having a `bindingset`-annotated predicate, I'd prefer to match the style of `parseArg` and `parseParam`.
```suggestion
private predicate parseField(string c, Field f) {
  specSplit(_, c, _) and
  exists(string splitRegex, string package, string className, string fieldName |
    fieldRegex = "^Field (.*)\\.([^.]+)\\.([^.]+)$" and
    package = c.regexpCapture(fieldRegex, 1) and
    className = c.regexpCapture(fieldRegex, 2) and
    fieldName = c.regexpCapture(fieldRegex, 3) and
    f.hasQualifiedName(package, className, fieldName)
  )
}
```